cxx-sensors/src/main/java/org/sonar/cxx/sensors/pclint/CxxPCLintSensor.java

```diff
@@ -109,21 +133,67 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
               }
 
               CxxReportIssue issue = new CxxReportIssue(id, file, line, msg);
-              saveUniqueViolation(context, issue);
+
+              if (currentIssue != null) {
+                saveUniqueViolation(context, currentIssue);
+              }
+
+              currentIssue = issue;
             } else {
               LOG.warn("PC-lint warning ignored: {}", msg);
               if (LOG.isDebugEnabled()) {
                 LOG.debug("File: {}, Line: {}, ID: {}, msg: {}", file, line, id, msg);
               }
             }
           }
+
+          if (currentIssue != null) {
+            saveUniqueViolation(context, currentIssue);
+          }
         } catch (com.ctc.wstx.exc.WstxUnexpectedCharException
           | com.ctc.wstx.exc.WstxEOFException
           | com.ctc.wstx.exc.WstxIOException e) {
           LOG.error("Ignore XML error from PC-lint '{}'", CxxUtils.getStackTrace(e));
         }
       }
 
+      private void addMoreLocationsToCurrentIssue(@Nonnull CxxReportIssue currentIssue,
+                                                String file,
+                                                String line,
+                                                String msg) {
+        if (file != null && file.isEmpty() && msg != null && msg.startsWith(PREFIX_DURING_SPECIFIC_WALK_MSG)) {
+          String walkedSrcFile =
+                  msg.substring(PREFIX_MSG_LENGTH, msg.indexOf(" ", PREFIX_MSG_LENGTH));
+
+          Matcher matcher = FILE_NAME_WITH_LINE_AND_COL_PATTERN.matcher(walkedSrcFile);

```

Comment:
@bloodlee why not including the prefix into the Regel?

Reply:
The pattern is only for extracting the filename, line number. And I think matching prefix only is faster.

Reply:
Hi @bloodlee,

think there is no difference in performance.
Now you have to:
- search for string
- create a new string (copy rest of line)
- search with regex again

Regex would break immediately after one sign does not fit (same as with your compare). For my understanding code would be much better to read.

Regards,

Reply:
You convinced me! I will commit the update.